go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/kubeflow/mpi-operator
 
-go 1.20
+go 1.22
```

Comment:
Could you submit the dedicated PR since all other PRs depend on this Go version update?